### ISSUE TITLE
No completion (after update?)

### ISSUE THREAD

**Issue by jD91mZM2** [racer-rust/vim-racer#92]
Today I decided to `:PlugUpdate`. Not sure if that's why, but I later today realized no completions at all worked.

It has worked previously, and my `init.vim` file looks like [this](https://github.com/legolord208/.dotfiles/blob/master/neovim/init.vim),
where it's worth to point out
```vim
call plug#begin()
Plug 'rust-lang/rust.vim'
Plug 'racer-rust/vim-racer'
call plug#end()
```

and

```vim
filetype plugin indent on

let g:racer_cmd = '~/.cargo/bin/racer'
let g:racer_experimental_completer = 1
```

No idea why it fails. Still doesn't work if I replace `~` with `/home/legolord208`, so it's not the expansion that is the problem.

EDIT:

> Still doesn't work

*accidentally said "Works" at first*

--------------------------------------------------------------------------

TL;DR `Omni completion (^O^N^P) Pattern not found`

**Comment by Shougo** [comment]
> No idea why it fails. Works if I replace ~ with /home/legolord208, so it's not the expansion that is the problem.

Hm.  The expansion is implemented in ftplugin/rust_racer.vim.
Do you disable the ftplugin?

https://github.com/racer-rust/vim-racer/blob/master/ftplugin/rust_racer.vim#L22

**Comment by Shougo** [comment]
@legolord208 Please test https://github.com/racer-rust/vim-racer/pull/93.

**Comment by jD91mZM2** [comment]
Sorry, I meant `Doesn't work`*. Sorry about that

**Comment by Shougo** [comment]
OK.  I will try it.
Please create the minimal init.vim less than 20 lines without the plugin manager configuration.
And please upload the reproduce ways from neovim starting.

We have no ESP skills.  I can only fix the reproduceable problem.

**Comment by Shougo** [comment]
And please check racer command works without vim-racer.

https://github.com/racer-rust/vim-racer/issues/90#issuecomment-301960100

**Comment by Shougo** [comment]
RUST_SRC_PATH is set?
I cannot finx it in your init.vim.

**Comment by jD91mZM2** [comment]
RUST_SRC_PATH is set. Racer commands work.

EDIT: I generally set environment variables in my `.profile`, that's why it's not in `init.vim`.

**Comment by Shougo** [comment]
Please upload the minimal init.vim and the reproduce ways.

**Comment by jD91mZM2** [comment]
```Rust
call plug#begin()
Plug 'rust-lang/rust.vim'
Plug 'racer-rust/vim-racer'
call plug#end()

let g:racer_cmd = '~/.cargo/bin/racer'
let g:racer_experimental_completer = 1
```

Literally just start vim and try to C-x C-o something.

**Comment by Shougo** [comment]
Please upload the reproduce ways.
It is required.

**Comment by jD91mZM2** [comment]
> Literally just start vim and try to C-x C-o something.

There is nothing else I know. Because doing that does not work for me.

**Comment by Shougo** [comment]
Please upload your racer and rust version.

**Comment by jD91mZM2** [comment]
```
> rustc --version
rustc 1.18.0 (03fc9d622 2017-06-06)
> racer --version
racer 2.0.8
```

**Comment by Shougo** [comment]
If `g:racer_experimental_completer` is 0, it works?

**Comment by jD91mZM2** [comment]
Nope

**Comment by Shougo** [comment]
Hm....
I will try it later.

**Comment by Shougo** [comment]
> Literally just start vim and try to C-x C-o something.

The buffer `&filetype` is rust?

**Comment by jD91mZM2** [comment]
Yep

**Comment by Shougo** [comment]
My check will take some time.
You can debug the omnifunc implementation.

https://github.com/racer-rust/vim-racer/blob/master/autoload/racer.vim#L13

**Comment by Shougo** [comment]
Especially, please test `cmd` from your command line and upload the result.

https://github.com/racer-rust/vim-racer/blob/master/autoload/racer.vim#L18

**Comment by jD91mZM2** [comment]
The code
23 `use std::`
and pressing C-x C-o says `Omni completion (^O^N^P) Pattern not found`.

Typing `racer complete 23 9 main.rs` spams me with `MATCH`es.
![image](https://user-images.githubusercontent.com/12830969/27170664-0561ae9c-51af-11e7-820d-2bce54f231c6.png)

**Comment by Shougo** [comment]
Please put `echomsg string(res)` result.
https://github.com/racer-rust/vim-racer/blob/master/autoload/racer.vim#L19

It may print the error.

**Comment by jD91mZM2** [comment]
Typing `:call system("racer 23 9 main.rs")` does absolutely nothing :thinking:

**Comment by Shougo** [comment]
`echo system("racer complete 23 9 main.rs")`

**Comment by Shougo** [comment]
And vim-racer uses tempfile.
So please test tempfile https://github.com/racer-rust/vim-racer/blob/master/autoload/racer.vim#L15.

**Comment by Shougo** [comment]
https://github.com/racer-rust/vim-racer/blob/master/autoload/racer.vim#L19

Do you know Vim script?
I think to add `echomsg string(res)` is better in the line.

**Comment by jD91mZM2** [comment]
I don't know a lot of vim script, sorry

EDIT: `echo system("racer complete 23 9 main.rs")` works just fine
EDIT 2: `echo tempfile()` does work :+1:

**Comment by Shougo** [comment]
Please insert `echomsg string(res)` in L20.
It is not hard work.

**Comment by jD91mZM2** [comment]
I did that in `~/.config/nvim/plugged/vim-racer/autoload/racer.vim` now. No difference when I press C-x C-o.

**Comment by Shougo** [comment]
`g:racer_experimental_completer` is 1?

**Comment by jD91mZM2** [comment]
Yes

**Comment by Shougo** [comment]
Please execute `:messages` command.
No errors?

I think it is errored.
https://github.com/racer-rust/vim-racer/blob/master/autoload/racer.vim#L260

**Comment by Shougo** [comment]
You can check the function behavior.
https://github.com/racer-rust/vim-racer/blob/master/autoload/racer.vim#L236

**Comment by Shougo** [comment]
Please check `verbose setlocal omnifunc`.

**Comment by jD91mZM2** [comment]
Oh my god `:messages` showed the error.
It's because my bashrc is set to print a message on startup. And then your plugin is getting that message unsure how to handle it.
LOL! It's not your plugin's fault, it's my bashrc's.

Thanks a lot for your help! :smile:

**Comment by Shougo** [comment]
Oh, OK...
